R/rename-files.R

```diff
@@ -8,124 +8,142 @@
-      i = "See {.help usethis::use_git} for help."
+      i = "See {.fn usethis::use_git} for help."
     ))
   }
-  # looking for the object name as well if changing from a file name to another
-  path_file_name <- fs::path_ext_remove(old)
-  file_name_base <- fs::path_file(path_file_name)
-  new_name_base <- fs::path_file(fs::path_ext_remove(new))
+  # After here, we start doing some renaming real situations---
+  renaming_strategy <- scope_rename(old, new, warn_conflicts)
 
-  # don't check for regexp if the original file name has less than min_n_char
-  min_n_char <- 5
-  cnd_check_for_object_names <-
-    file_name_base != new_name_base &&
-    !file_name_base %in% (c("index", "temp")) &&
-    nchar(file_name_base) > min_n_char &&
-    !tolower(fs::path_ext(old)) %in% c("png", "jpg", "jpeg", "pdf", "svg") # don't check for figures
+  regexp_to_search_for_in_files <- compute_conflicts_regex(old, renaming_strategy)
 
-  if (cnd_check_for_object_names) {
-    object_snake_from_file_kebab <- stringr::str_replace_all(file_name_base, "-", "_")
-    regex_file_name <- paste0(c(object_snake_from_file_kebab, old), collapse = "|")
-  } else {
-    regex_file_name <- paste0(path_file_name, "[^-]?")
-  }
-
-  related_files <- fs::dir_ls(regexp = paste0(regex_file_name, "\\."), recurse = TRUE)
+  # FIXME doesn't fit now.
+  related_files <- fs::dir_ls(regexp = paste0(basename_remove_ext(old), "\\."), recurse = TRUE)
   related_files <- setdiff(related_files, old)
   if (length(related_files) > 0) {
     cli::cli_warn(c(
       "Other files have a similar pattern",
       "See {.file {related_files}}",
-      "No support yet for that"
+      "No support yet for that yet.",
+      "Think about what triggers this and add new rules  in {.fn scope_rename}"
     ))
   }
-  if (!force) {
+
+
+  if (renaming_strategy != "free_for_all") {
     extra_msg_if_file_conflict <- c(
       x = "Did not rename files!",
       "!" = paste0("Found references to {.val ", old, "} in project"),
-      i = paste0("Change file path to {.val ", new, "} or see {.run [Find in Files](rstudioapi::executeCommand('findInFiles'))} Replace All if confident. {.emph Copied new name to clipboard}"),
-      i = "Also change object names to snake_case that follow the new file name."
+      i = paste0("Change file path to {.val ", new, "} in files ahead of renaming file or \\
+                  see {.run [Find in Files](rstudioapi::executeCommand('findInFiles'))} Replace All if confident. {.emph Copied new name to clipboard}"),
+      if (!is_moving(old, new)) i <- "Also change object names to snake_case that follow the new file name."
     )
   } else {
     extra_msg_if_file_conflict <- c("Here are the conflicts. Review changes carefully", "renaming file anyway")
   }
 
-  verbose <- cnd_check_for_object_names | length(related_files) > 0 | force
+  verbose <- TRUE # length(related_files) > 0 || renaming_strategy == "free_for_all"
```

Comment:
Not sure I got verbosity right, verify with snapshots.

I think that we want to increase verbosity when we set `renaming_strategy == "free_for_all"` , action = "test", overwrite = TRUE.

Validate with snapshots..

Reply:
Maybe should deprecate action = "test".. no longer feels useful.

Instead of force, just use `file_move()`..